routines/metric_pruning.go

```diff
@@ -31,7 +31,7 @@ type MetricPruningRoutine struct {
 	interval                     time.Duration
 	startDelay                   time.Duration
 	maxRequestMetricsHistoryDays int64
-	*database.PostgresClient
+	Database                     database.MetricsDatabase
```

Comment:
Doesn't seem to need to be public? & to keep consistent with the other routines

```suggestion
	db                     database.MetricsDatabase
```